File: csharp/NumberOfSegmentsInAString.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LeetCode
{
    class NumberOfSegmentsInAString
    {
        public int CountSegments(string s)
        {
            int numSegments = 0;
            for (int i = 0; i < s.Length; i++)
            {
                if ((i == 0 || s[i - 1] == ' ') && s[i] != ' ')
                {
                    numSegments += 1;
                }
            }
            return numSegments;
        }
    }
}

File: js/NumberOfSegmentsInAString.js
/**
 * @param {string} s
 * @return {number}
 */
var countSegments = function(s) {
    let numSegments = 0;
    for(let i = 0; i < s.length; i++){
        if((i === 0 || s.charAt(i-1) === ' ') && s.charAt(i) !== ' '){
            numSegments += 1;
        }
    }
    return numSegments;
};